sql/updates/pre-global.sql

```diff
@@ -1,3 +1,10 @@
---need to keep this legacy functions around for 0.9 since it's still used in 0.8, delete after 0.9
-CREATE OR REPLACE FUNCTION _timescaledb_internal.ddl_command_end() RETURNS event_trigger
-AS '@MODULE_PATHNAME@', 'timescaledb_ddl_command_end' LANGUAGE C;
+-- Triggers should be disabled during upgrades to avoid having them
+-- invoke functions that might load an old version of the shared
+-- library before those functions have been updated.
+DROP EVENT TRIGGER IF EXISTS timescaledb_ddl_command_end;
```

Comment:
So I find the semantics here to be weird: we drop all these triggers in pre-global but they are added back at random points in the script so some update command will still execute after triggers are again created. Should we move the creation of all triggers to post-global or something?

Reply:
Not sure what the problem is really. Can you elaborate? What this ensures is that triggers aren't invoked until they reference the right shared library.

Reply:
So look I think there are two semantics we may want:
1) Triggers are disabled during updates - in which case we should drop the triggers in the pre-global and add them back in post-global. That way we know they are disabled throughout the /entire/ upgrade script.

2) Be sure that the trigger always references the function in the right shared library. In which case the most direct way of doing that is to put the trigger function definition in pre-global instead of dropping the trigger. 

As it is I think we have a weird mixture of the two. I'd rather have a cleaner semantics one way or the other.

Reply:
What you suggest would surely work, but I don't see that it provides much over what is present apart from making things more complicated. The semantics we care about is that triggers are disabled until the right trigger function is installed, and that is exactly what happens now.

If you `CREATE` the extension, the same thing applies. The triggers will be created and enabled at some point during the process (depending on where included) and, obviously, the trigger functions need to be defined before the triggers are defined. But it might not be at the very end. If this works for `CREATE` it should also work for updates since we add everything `CREATE` installs at the end of the update.

If you feel strongly about this then I can do it, but I don't see why we should add extra files and ordering requirements on these files without that approach actually providing any tangible benefits, all things considered. 

I would see the benefit of (2) if we would require that triggers are available during updates, but otherwise I don't see the point.

Note, also, that updates will fail in testing if triggers for some reason would reference the wrong lib. So, if someone screws this up, it will easily be detectable if we get the tests in shape.

Finally, if you'd look at how much simpler and clearer the `CMakeLists.txt` is now, and how many fewer files we have (no pre,post files, etc.), then maybe you'll see that adding back a lot of complexity there is not really worth it. Maybe we can add back that complexity if we find that we really need it?